File: 347-top-k-frequent-elements/top-k-frequent-elements.py
from collections import defaultdict

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        nums_map = defaultdict(int)

        for num in nums:
            nums_map[num] += 1

        sort_dict = dict(sorted(nums_map.items(), key=lambda item: item[1], reverse=True))

        keys = list(sort_dict)

        return keys[:k]

File: 347-top-k-frequent-elements/top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> freqCntMp;
        vector<vector<int>> v(nums.size()+1, vector<int>());
        int maxSize = 0;
        for(int i=0;i<nums.size();i++){
            freqCntMp[nums[i]]+=1;
            v[freqCntMp[nums[i]]].push_back(nums[i]);
            if(maxSize<freqCntMp[nums[i]])
                maxSize = freqCntMp[nums[i]];
        }
        for(int idx = maxSize;idx>0;idx--){
            if(v[idx].size()==k){
                return v[idx];
            }
        }
        return vector<int>();
    }
};